body.tex

```diff
@@ -107,8 +107,16 @@ \subsection{Registering}
 \end{itemize}
 \validation
 \begin{itemize}
-    \item $\field{O}{id}$ must be unique, \ie not currently in use,
-        between $1$ and $32$ bytes long, and valid \textsf{UTF-8}.
+    \item $\field{O}{id}$ must be unique, \ie not currently in use as an org
+        identifier {\em or} user identifier, and comply with the following
+        rules:
+        \begin{itemize}
+            \item Must be between $1$ and $32$ bytes long.
+            \item Must only contain \texttt{a-z}, \texttt{0-9} and `\texttt{-}'
+                characters.
+            \item Must not start or end with a `\texttt{-}'.
+            \item Must not contain sequences of more than one `\texttt{-}'.
```

Comment:
Since we are thinking about this, do we want identifiers starting and or ending with `-`, such as `-abc-`, `-abc`, `abc-`? Haven't checked if other systems allow that but it seems unlikely they would.

Reply:
Line `117` says that it cannot start or end with `-`

Reply:
Missed that one!